### PR TITLE
Implement category item card

### LINKED ISSUE #175: (SP: 1) Implement category item card
UserStory: #52, #54
## Description
Implement category card so there will be icon, title, description (amount of offers for this category). Also, there should be a link so if a user clicks on the card a page should redirect to it.
## Design 
Image
left - hovered, right - normal
## Resources
Links to the documentations, libraries, guides 
## Acceptance Criteria
Can be copied from user story or give more detailed, we can use 
- [ ] the block should be responsive to hover
- [ ] when a user clicks on the card the should be redirected to "Subjects" page with selected category

### PR DESCRIPTION
**Pull request by YaroslavChuiko**
- [x] the block should be responsive to hover
- [x] when a user clicks on the card the should be redirected to "Subjects" page with selected category

image
image

### CODE REVIEW

File: src/components/category-card/CategoryCard.styles.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        mb: '4px'
+      },
+      description: {
+        fontSize: '14px',
+        color: 'primary.500',
+        typography: 'subtitle2',
+        fontWeight: 400
+      }
+    },
+    root: {
```
Comment (khytrykv) [inline src/components/category-card/CategoryCard.styles.js:23]: Where do we use `root` styles?

**Review by khytrykv — APPROVED** [review]

**Review by OlhaOvsiienko — APPROVED** [review]

File: src/components/category-card/CategoryCard.styles.js (hunk 1 above)
Comment (YaroslavChuiko) [reply src/components/category-card/CategoryCard.styles.js:23]: I pass it to the ImgTitleDescription component

image

**Review by YaroslavLys — APPROVED** [review]

File: tests/unit/components/category-card/CategoryCard.spec.jsx
Comment (vlad-khrychov) [inline tests/unit/components/category-card/CategoryCard.spec.jsx:1]: Add test for the routing. 
https://medium.com/@maksimovic.danijel989/how-to-test-react-router-components-1b399c89ae3c

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]

File: tests/unit/components/category-card/CategoryCard.spec.jsx
Comment (vlad-khrychov) [inline tests/unit/components/category-card/CategoryCard.spec.jsx:1]: Mock `ImgTitleDescription` and I think you will get 100%.

**Review by vlad-khrychov — CHANGES_REQUESTED** [review]

**Review by vlad-khrychov — APPROVED** [review]

**Pull request merged**